lib/jekyll-optional-front-matter/generator.rb

```diff
@@ -58,5 +68,17 @@ def markdown_converter
     def entry_filter
       @entry_filter ||= Jekyll::EntryFilter.new(site)
     end
+
+    def option(key)
+      site.config[CONFIG_KEY] && site.config[CONFIG_KEY][key]
+    end
+
+    def disabled?
+      option(DISABLED_KEY) || site.config["require_front_matter"]
```

Comment:
Nice! Should we throw a deprecated warning via `Jekyll.logger.warn`?

Reply:
I don't think it is worth bothering people with this. This option could be part of the spec for the entire lifetime of this plugin without causing much trouble.